### ISSUE TITLE
Use compressors collection library

### ISSUE THREAD

**Issue by jakirkham** [dask/distributed#6265]
In PR ( https://github.com/dask/distributed/pull/6259 ), it was suggested that we might benefit from using one library that collects several compressors. Some libraries like this include [cramjam]( https://github.com/milesgranger/pyrus-cramjam ), [compress]( https://github.com/MacHu-GWU/compress-project ), [numcodecs]( https://github.com/zarr-developers/numcodecs ), [python-libarchive-c]( https://github.com/Changaco/python-libarchive-c ), etc.. This would allow us to have one dependency to use for our compressors.

**Comment by jakirkham** [comment]
cc @martindurant

**Comment by jakirkham** [comment]
Yeah the decompressing into a buffer feature can be useful. Not sure if we need that in Distributed serialization currently, but that could change. Though I think with that we need to track the decompressed buffer size separately right? Or does cramjam add that to the compressed result somehow (perhaps as a binary header)?

**Comment by martindurant** [comment]
In the parquet case, we know the decompressed size - otherwise you have to send it separately. I think maybe one or two of the codecs contain the size in a header.

**Comment by jakirkham** [comment]
Yeah we could certainly add logic to track buffer sizes pre-compression. It's not hard. There may be some value in doing that.
